compiler/codegen/codegen.cpp

```diff
@@ -2509,9 +2578,49 @@ void makeBinary(void) {
     mysystem(command, "compiling generated source");
   }
 
-  if (fLibraryCompile && fLibraryPython) {
-    codegen_make_python_module();
+  if (gCodegenGPU == false) {
+    if (fLibraryCompile && fLibraryPython) {
+      codegen_make_python_module();
+    }
+  }
+}
+
+void codegen() {
+  if (no_codegen)
+    return;
+
+  codegenPartOne();
+
+#ifdef HAVE_LLVM
```

Comment:
I don't see why this code should be in the ifdef here.

Reply:
When compiling chapel with `CHPL_LLVM=none`, I get an error
```
codegen.cpp: In function ‘void codegen()’:
../include/misc.h:35:17: error: ‘dont_use_exit_use_clean_exit_instead’ was not declared in this scope
 #define exit(x) dont_use_exit_use_clean_exit_instead
                 ^
../include/misc.h:35:17: note: in definition of macro ‘exit’
 #define exit(x) dont_use_exit_use_clean_exit_instead
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ````
`codegen()` uses `exit(0)` to exit the child process created to do the gpu code generation. Without the ifdef and for `CHPL_LLVM=none`, `exit(x)` is defined with ```dont_use_exit_use_clean_exit_instead```, which doesn't exist anymore. I added the ifdef since gpu code generation is currently only done when llvm is used by the chapel compiler and when llvm is being used exit(x) is defined with `#define exit(x) clean_exit(x)` instead.

Reply:
AFAIK you can replace the `exit(0)` with `clean_exit(0)` - have you tried that?